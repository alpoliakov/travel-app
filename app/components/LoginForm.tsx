import Link from 'next/link';
import { useRouter } from 'next/router';
import React, { useEffect, useState } from 'react';
import { useIntl } from 'react-intl';

import { REG_EMAIL } from '../constants';
import { useAuth } from '../lib/useAuth';
import Loader from './Loader';

export default function LoginForm() {
  const [userData, setUserData] = useState({
    email: '',
    password: '',
  });

  const router = useRouter();
  const [loading, setLoading] = useState(false);
  const [errorPsw, setErrorPsw] = useState(false);
  const [validEmail, setValidEmail] = useState(false);

  const { signIn, user, error } = useAuth();

  const { formatMessage: f } = useIntl();

  const onFinish = (e) => {
    e.preventDefault();
    setLoading(true);
    const { email, password } = userData;

    if (password.length < 6) {
      setErrorPsw(true);
    }

    if (email.match(REG_EMAIL)) {
      setValidEmail(false);
    } else {
      setValidEmail(true);
    }

    signIn(email, password);
  };

  useEffect(() => {
    if (user) {
      setLoading(false);
    }

    if (error) {
      setLoading(false);
    }
  }, [user, error]);

  return (
    <div className="w-full max-w-lg">
      <div className="leading-loose">
        <Loader show={loading} />
        <form className="m-4 p-10 bg-white bg-opacity-25 rounded shadow-xl">
          <p className="text-white font-medium text-center text-lg font-bold uppercase">
            {f({ id: 'btnLogin' })}
          </p>
          <div className="">
            <label className="block text-sm text-white" htmlFor="email">
              E-mail:
            </label>
            <input
              className="w-full px-5 py-1 text-gray-700 bg-gray-300 rounded focus:outline-none focus:bg-white"
              type="email"
              id="email"
              value={userData.email}
              placeholder={f({ id: 'phEmail' })}
              aria-label="email"
              onChange={(e) => setUserData({ ...userData, email: e.target.value })}
              required
            />
            <span className="flex items-center font-medium tracking-wide text-yellow-500 text-xs mt-1 ml-1">
              {validEmail && f({ id: 'errorEmail' })}
            </span>
          </div>
          <div className="mt-4">
            <label className="block  text-sm text-white" htmlFor="password">
              {f({ id: 'pass' })}
            </label>
            <input
              className="w-full px-5 py-1 text-gray-700 bg-gray-300 rounded focus:outline-none focus:bg-white"
              type="password"
              id="password"
              value={userData.password}
              onChange={(e) => {
                setUserData({ ...userData, password: e.target.value });
                if (userData.password.length > 5) {
                  setErrorPsw(false);
                }
              }}
              name="password"
              placeholder={f({ id: 'phPass' })}
              arial-label="password"
              required
            />
            <span className="flex items-center font-medium tracking-wide text-yellow-500 text-xs mt-1 ml-1">
              {errorPsw && f({ id: 'errorPass' })}
            </span>
          </div>

          <div className="mt-8 items-center flex justify-between">
            <button
              className="px-4 py-1 text-white font-bold tracking-wider bg-gray-900 hover:bg-gray-800 rounded"
              type="submit"
              onClick={onFinish}>
              {f({ id: 'enter' })}
            </button>
            <p
              className="inline-block cursor-pointer right-0 align-baseline font-bold text-sm text-500 text-white hover:text-blue-800"
              role="presentation"
              onClick={() => router.back()}>
              {f({ id: 'return' })}
            </p>
          </div>
          <div className="text-center">
            <Link href="/auth/signup">
              <a className="inline-block right-0 align-baseline font-bold text-sm text-500 text-white hover:text-blue-800">
                {f({ id: 'createAcc' })}
              </a>
            </Link>
          </div>
        </form>
      </div>
    </div>
  );
}
