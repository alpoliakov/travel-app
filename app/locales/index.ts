export const messages = {
  en: {
    btnLogin: 'Login',
    btnSignUp: 'Sign Up',
    search: 'Search',
    phEmail: 'Enter your e-mail',
    phPass: 'Enter your password',
    createAcc: 'Create an account',
    forgetPass: 'Forgot password ?',
    enter: 'Enter',
    load: 'Loading',
    high: 'Max',
    low: 'Min',
    now: 'Currently',
    pass: 'Password (6 characters minimum):',
    phName: 'Enter your name',
    phAvatar: 'Upload your avatar',
    name: 'Name (3 characters minimum):',
    avatar: 'Avatar:',
    register: 'Sign Up',
    signOut: 'Sign Out',
    errorEmail: 'Invalid email field!',
    errorPass: 'Invalid password field!',
    errorName: 'Invalid username field!',
    profile: 'Profile',
    delete: 'Delete',
    or: 'Or',
    selectFile: 'Select a file',
    hello: 'Hello!',
    modalText: 'To view, you need to log in or register!',
    close: 'Close',
  },
  ru: {
    btnLogin: 'Войти',
    btnSignUp: 'Регистрация',
    search: 'Поиск',
    phEmail: 'Введите ваш e-mail',
    phPass: 'Введите ваш пароль',
    createAcc: 'Создать аккаунт',
    forgetPass: 'Забыли пароль ?',
    enter: 'Ввести',
    load: 'Загрузка',
    high: 'Макс',
    low: 'Мин',
    now: 'Сейчас',
    pass: 'Пароль (6 символов минимум):',
    phName: 'Введите ваше имя',
    phAvatar: 'Загрузите ваш аватар',
    name: 'Имя (3 символа минимум):',
    avatar: 'Аватар:',
    register: 'Регистрация',
    signOut: 'Выйти',
    errorEmail: 'Невалидное поле электронной почты!',
    errorPass: 'Невалидное поле пароля!',
    errorName: 'Невалидное поле имени пользователя!',
    profile: 'Профиль',
    delete: 'Удалить',
    or: 'Или',
    selectFile: 'Выберите файл',
    hello: 'Приветствуем!',
    modalText: 'Для просмотра необходимо авторизоваться или зарегистрироваться!',
    close: 'Закрыть',
  },
  uk: {
    btnLogin: 'Увійти',
    btnSignUp: 'Реєстрація',
    search: 'Пошук',
    phEmail: 'Введіть ваш e-mail',
    phPass: 'Введіть ваш пароль',
    createAcc: 'Створити акаунт',
    forgetPass: 'Забули пароль ?',
    enter: 'Ввести',
    load: 'Завантаження',
    high: 'Макс',
    low: 'Miн',
    now: 'Зараз',
    pass: 'Пароль (6 символів мінімум):',
    phName: 'Введіть ваше им`я',
    phAvatar: 'Завантажте ваш аватар',
    name: 'Ім`я (3 символа мінімум):',
    avatar: 'Аватар:',
    register: 'Реєстрація',
    signOut: 'Вийти',
    errorEmail: 'Невалідне поле електронної пошти!',
    errorPass: 'Невалідне поле пароля!',
    errorName: 'Невалідне поле імені користувача',
    profile: 'Профіль',
    delete: 'Видалити',
    or: 'Або',
    selectFile: 'Виберіть файл',
    hello: 'Вітаємо!',
    modalText: 'Для перегляду потрібно увійти або зареєструватися!',
    close: 'Закрити',
  },
};
