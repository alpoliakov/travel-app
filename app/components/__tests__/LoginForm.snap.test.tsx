import React from 'react';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';

import { messages } from '../../locales';
import LoginForm from '../LoginForm';

jest.mock('../../lib/useAuth', () => ({
  useAuth: jest.fn(() => ({ signIn: jest.fn(), user: '', error: '' })),
  AuthProvider: ({ children }) => <div>{children}</div>,
}));

jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'),
  useRouter: () => ({
    route: '/',
    pathname: '',
    query: '',
    asPath: '',
    locale: 'en',
    locales: ['en', 'ru', 'uk'],
  }),
}));

it('LoginForm renders correctly', () => {
  const tree = renderer
    .create(
      <IntlProvider locale="ru" messages={messages['ru']}>
        <LoginForm />)
      </IntlProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
