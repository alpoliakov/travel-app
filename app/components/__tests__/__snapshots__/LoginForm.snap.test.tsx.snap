// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
Array [
  <div
    className="w-full max-w-lg"
  >
    <div
      className="leading-loose"
    >
      <form
        className="m-4 p-10 bg-white bg-opacity-25 rounded shadow-xl"
      >
        <p
          className="text-white font-medium text-center text-lg font-bold uppercase"
        >
          Войти
        </p>
        <div
          className=""
        >
          <label
            className="block text-sm text-white"
            htmlFor="email"
          >
            E-mail:
          </label>
          <input
            aria-label="email"
            className="w-full px-5 py-1 text-gray-700 bg-gray-300 rounded focus:outline-none focus:bg-white"
            id="email"
            onChange={[Function]}
            placeholder="Введите ваш e-mail"
            required={true}
            type="email"
            value=""
          />
          <span
            className="flex items-center font-medium tracking-wide text-yellow-500 text-xs mt-1 ml-1"
          />
        </div>
        <div
          className="mt-4"
        >
          <label
            className="block  text-sm text-white"
            htmlFor="password"
          >
            Пароль (6 символов минимум):
          </label>
          <input
            arial-label="password"
            className="w-full px-5 py-1 text-gray-700 bg-gray-300 rounded focus:outline-none focus:bg-white"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Введите ваш пароль"
            required={true}
            type="password"
            value=""
          />
          <span
            className="flex items-center font-medium tracking-wide text-yellow-500 text-xs mt-1 ml-1"
          />
        </div>
        <div
          className="mt-8 items-center flex justify-between"
        >
          <button
            className="px-4 py-1 text-white font-bold tracking-wider bg-gray-900 hover:bg-gray-800 rounded"
            onClick={[Function]}
            type="submit"
          >
            Ввести
          </button>
          <p
            className="inline-block cursor-pointer right-0 align-baseline font-bold text-sm text-500 text-white hover:text-blue-800"
            onClick={[Function]}
            role="presentation"
          >
            Вернуться ?
          </p>
        </div>
        <div
          className="text-center"
        >
          <a
            className="inline-block right-0 align-baseline font-bold text-sm text-500 text-white hover:text-blue-800"
            href="/auth/signup"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Создать аккаунт
          </a>
        </div>
      </form>
    </div>
  </div>,
  ")",
]
`;
